src/bot.py

```
@@ -5,785 +5,744 @@
+
+                                if is_dm:
+                                    is_expected_author = prev.author in (self.safe_user, current_msg.author)
+                                else:
+                                    is_expected_author = prev.author == current_msg.author
+
+                                if is_valid_type and is_expected_author:
+                                    next_msg = prev
+                                break
+                        current_msg = next_msg
+                    except (discord.NotFound, discord.HTTPException):
+                        logging.exception(f"Failed to fetch parent for message {current_msg.reference.message_id}")
```

Ruff (0.14.10)

`exception()` call on root logger

(LOG015)